auth/src/main/java/org/aerogear/auth/ClientRole.java

```diff
@@ -1,7 +1,41 @@
 package org.aerogear.auth;
 
+/**
+ * Creates a new keycloak client role for a user
+ */
 public final class ClientRole extends AbstractRole {
```

Comment:
this is framework code, right? be careful w/ making things final... If they not really really really need to be final

Reply:
It is like that by purpose. The idea is that a ClientRole is a ClientRole: we don't want it to be customized. If another role should be created, it should extend AbstractRole, so that the framework know that a ClientRole behave in a certain way and can't be changed.

Reply:
than, let's state that, why it's locked down.

and why we don't think that a developer should not extend `ClientRole` for his `SuperDopeClientRole`. 

I've seen lot's of _traffic_ on mailing lists before that certains things, in OSS frameworks, were locked/final.  

Sure, delegation is possible - but stinks :)